libr/asm/asm.c

```diff
@@ -1110,15 +1110,15 @@ R_API RAsmCode* r_asm_rasm_assemble(RAsm *a, const char *buf, bool use_spp) {
 	if (use_spp) {
 		Output out;
 		out.fout = NULL;
-		out.cout = r_strbuf_new ("");
-		r_strbuf_init (out.cout);
+		out.cout = (SStrBuf *)r_strbuf_new ("");
+		r_strbuf_init ((RStrBuf *)out.cout);
 		struct Proc proc;
 		spp_proc_set (&proc, "spp", 1);
 
 		lbuf = replace_directives (lbuf);
 		spp_eval (lbuf, &out);
 		free (lbuf);
-		lbuf = strdup (r_strbuf_get (out.cout));
+		lbuf = strdup (r_strbuf_get ((RStrBuf *)out.cout));
```

Comment:
i would like to not merge those casts. the correct fix is to make spp use r_util when built inside r2, not to cast like this and in the future get segfaults without much hints to understand the reason. can you drop those casts from the pr? and i'll merge?  ill try to find some time to fix this spp warnings properly before sunday

Reply:
sure